src/ide/tgui.ts

```diff
@@ -130,6 +133,13 @@ export let tgui = (function () {
 			element.tabIndex = description.tabindex;
 		}
 
+		// html tag attributes, TODO: should these be in the description
+		// object directly?
+		if (description.hasOwnProperty("attributes")) {
+			for (let a in description.attributes)
+				element.setAttribute(a, description.attributes[a]);
+		}
+
```

Comment:
Did you use this in your implementation of dark mode?

Reply:
I have used it when I changed the layout of the check boxes. Before, the description object could not hold various html attributes, because those would simply be ignored in this function. So I added an explicit attributes tag, which contains html attributes that would definitely be applied to the element. Maybe it wouldn't be a bad idea to move them back in to the enclosing description object, but then it might be difficult to differenciate between custom description attributes and html attributes.